src/freenas/usr/local/lib/middlewared_truenas/plugins/enclosure_/map.py

```diff
@@ -120,11 +140,15 @@ async def _map_enclosures(self, enclosures, slots):
             elements.append(element)
 
         info = await self.middleware.call("system.info")
```

Comment:
`system.info` actually runs a bunch of unnecessary/unrelated API calls that are irrelevant for this. `system.dmidecode_info` is much better suited for this particular scenario.

Reply:
That's probably a good fix but don't want to mix unrelated fixes